SingularityService/src/test/java/com/hubspot/singularity/mesos/SingularityMesosTaskBuilderTest.java

```diff
@@ -138,6 +141,40 @@ public void testJobUserPassedAsEnvironmentVariable() {
   }
 
   @Test
+  public void testEnvironmentVariableOverrides() {
+    Map<String, String> overrideVariables = new HashMap<>();
+    overrideVariables.put("HS_ACCESS_GROUP", "test");
+    overrideVariables.put("STARTED_BY_USER", "notTestUser");
+
+    SingularityRunNowRequest runNowRequest = new SingularityRunNowRequestBuilder()
+        .environmentVariables(overrideVariables)
+        .build();
+
+    final SingularityRequest request = new SingularityRequestBuilder("test", RequestType.WORKER)
+        .build();
+    final SingularityDeploy deploy = new SingularityDeployBuilder("test", "1")
+        .setCommand(Optional.of("/bin/echo hi"))
+        .setRunImmediately(Optional.of(runNowRequest))
+        .build();
+    final SingularityTaskRequest taskRequest = new SingularityTaskRequest(request, deploy, pendingTask);
+    final SingularityTask task = builder.buildTask(offerHolder, null, taskRequest, taskResources, executorResources);
+
+    Map<String, String> environmentVariables = task.getMesosTask()
```

Comment:
could probably just do the for loop here. `for (Variable var : task.getMesosTask().getCommand().getEnvironment().getVariablesList())` and just do the assert equals against getName/getKey. Saves you streaming to a map, only to then iterate.

Reply:
True, but the mesos task variables are a superset of the override variables, so the loop would require some kind of checking like this: ```    Boolean checkedMyNewEnvVar = false;
    Boolean checkedStartedByUser = false;

    for (Variable var : task.getMesosTask().getCommand().getEnvironment().getVariablesList()) {
      if (overrideVariables.containsKey(var.getName())) {
        if (var.getName().equals("MY_NEW_ENV_VAR")) {
          checkedMyNewEnvVar = true;
        } else if (var.getName().equals("STARTED_BY_USER")) {
          checkedStartedByUser = true;
        }

        assertEquals(
            "Environment variable " + var + " not overridden.",
            var.getValue(),
            overrideVariables.get(var.getName()));
      }
    }
    assertTrue(checkedMyNewEnvVar);
    assertTrue(checkedStartedByUser);```
Which do you think is cleaner (or is there an even cleaner alternative?)?